functions/src/api.ts

```diff
@@ -1,28 +1,33 @@
 import * as admin from 'firebase-admin'
 import * as functions from 'firebase-functions'
 import * as Cors from 'cors'
+import { z } from 'zod'
 
 import { User, PrivateUser } from '../../common/user'
 import {
   CORS_ORIGIN_MANIFOLD,
   CORS_ORIGIN_LOCALHOST,
 } from '../../common/envs/constants'
 
+type Output = Record<string, unknown>
 type Request = functions.https.Request
 type Response = functions.Response
-type Handler = (req: Request, res: Response) => Promise<any>
 type AuthedUser = [User, PrivateUser]
+type Handler = (req: Request, user: AuthedUser) => Promise<Output>
 type JwtCredentials = { kind: 'jwt'; data: admin.auth.DecodedIdToken }
 type KeyCredentials = { kind: 'key'; data: string }
 type Credentials = JwtCredentials | KeyCredentials
 
 export class APIError {
   code: number
   msg: string
-  constructor(code: number, msg: string) {
+  details: unknown
```

Comment:
Details should be either a generic type or any. (Unknown should be used very sparingly in Typescript.)

Reply:
Can you clarify your criticism of the `unknown` type in these two cases? They seem conceptually correct to me. The intent is for them to represent
- any Javascript object
- which can be literally anything and we don't care what, because we are just a middleman echoing them through to `JSON.stringify`.